Module-08 Visualforce Workshop/classes/shoppingCartController.cls

```diff
@@ -0,0 +1,398 @@
+public class shoppingCartController 
+{
+    public boolean error {get;set;}
+    public String sortDirection {get;set;}
+    private Integer totalRecs1 = 0;
+    private Integer OffSetSize1 = 0;
+    private Integer LimitSize1 = 10;
+    private Integer totalRecs2 = 0;
+    private Integer OffSetSize2 = 0;
+    private Integer LimitSize2 = 10;
+    public String idOfItem {get;set;}
+    public String idOfItemQuant {get;set;}
+    public List<Id> productIds = new List<Id>();
+    public List<Product2> shoppingProduct {get;set;}
+    public String searchKey {get;set;}
+    
+    public List<Product2> orderItems 
+    {
+        get
+        {
+            if(ossc != null)
+            	return ossc.getRecords();
+            else return null;
+        } set;
+    }
+    
+    public List<Product2> checkoutItems 
+    {
+        get
+        {
+            if(cssc != null)
+                return cssc.getRecords();
+            else return null;
+        } set;
+    }
+    
+    public List<Purchase_Order__c> purchaseOrderHistory{get; set;}
+    public boolean hideShowCartList {get;set;}
+    public boolean hideShowCheckoutList {get;set;}
+    public boolean hideShowProductList {get;set;}
+    public decimal totalMoney {get;set;}
+    public product2 valueOfQuant;
+    
+    public ApexPages.StandardSetController ossc{set;get;}
+    
+    public Boolean hasNext1
+    {
+        get 
+        {
+            if(ossc != NULL)
+                return ossc.getHasNext();
+            return false;
+        } set;
+    }
+    
+    public Boolean hasPrev1
+    {
+        get
+        {
+            if(ossc != NULL) 
+                return ossc.getHasPrevious();
+            return false;
+        } set;
+    }
+    
+    public ApexPages.StandardSetController cssc{set;get;}
+    public Boolean hasNext2
+    {
+        get
+        {
+            if(cssc != NULL)
+                return cssc.getHasNext();
+            return false;
+        } set;
+    }
+    
+    public Boolean hasPrev2
+    {
+        get
+        {
+            if(cssc != NULL) 
+                return cssc.getHasPrevious();
+            return false;
+        } set;
+    }
+    
+    public boolean getHideShowCartList(boolean input)
+    {
+        hideShowCartList = input;
+        
+        return hideShowCartList;
+    }
+    
+    public boolean getHideShowCheckoutList(boolean input)
+    {
+        hideShowCheckoutList = input;
+        
+        return hideShowCheckoutList;
+    }
+    
+    public boolean getHideShowProductList(boolean input)
+    {
+        hideShowProductList = input;
+        
+        return hideShowProductList;
+    }
+    
+    public shoppingCartController()
+    {
+        totalMoney = 0.0;
+        error = false;
+        sortDirection = 'asc';
+        List<Product2> orderItems = new List<Product2>();
+        PurchaseOrderHistory = [SELECT Id, Order_Price__c, Order_Status__c FROM Purchase_Order__c LIMIT 5];
```

Comment:
too many queries. all over the place. too hard to manage and maintain plus what's coming from where hard to know.